Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java

```diff
@@ -133,39 +133,49 @@ public void clear() {
     }
 
     void deleteSelectedStructure() {
-        Optional<IncludedStructuralElement> selectedStructure = getSelectedStructure();
-        if (!selectedStructure.isPresent()) {
+        if (!getSelectedStructure().isPresent()) {
             /*
              * No element is selected or the selected element is not a structure
              * but, for example, a media unit.
              */
             return;
         }
-        LinkedList<IncludedStructuralElement> ancestors = MetadataEditor.getAncestorsOfStructure(selectedStructure.get(), structure);
+        IncludedStructuralElement selectedStructure = getSelectedStructure().get();
+        LinkedList<IncludedStructuralElement> ancestors = MetadataEditor.getAncestorsOfStructure(selectedStructure, structure);
         if (ancestors.isEmpty()) {
             // The selected element is the root node of the tree.
             return;
         }
         IncludedStructuralElement parent = ancestors.getLast();
 
         Collection<View> subViews = new ArrayList<>();
-        subViews = getAllSubViews(selectedStructure.get(), subViews);
+        getAllSubViews(selectedStructure, subViews);
+
+        List<View> multipleViews = subViews.stream().filter(v -> v.getMediaUnit().getIncludedStructuralElements().size() > 1)
+                .collect(Collectors.toList());
+        for (View view : multipleViews) {
```

Comment:
You are mixing recursion, a stream and iterating here. If you want to use streams, you can do all of the actions in one stream here, I think:
```java
Workpiece.treeStream(selectedStructure)
    .flatMap(logicalDivision -> logicalDivision.getViews().parallelStream()) // subViews
    .filter(view -> 
        {   List<IncludedStructuralElement> logicalDivisions = view.getMediaUnit().getIncludedStructuralElements();
            if (logicalDivisions.size() <= 1) {
                return true; // add to parent views
            }
            dataEditor.unassignView(selectedStructure, view, selectedStructure.getViews().getLast().equals(view));
            if (logicalDivisions.size() <= 1) {
                severalAssignments.remove(view.getMediaUnit());
            }
            return false; // don't add to parent views
        }
    )
    .forEachOrdered(parent.getViews()::add)
;
```